android/java/org/chromium/chrome/browser/rewards/BraveRewardsTipConfirmationFragment.java

```diff
@@ -0,0 +1,186 @@
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.fragment.app.Fragment;
+
+import org.chromium.base.Log;
+import org.chromium.chrome.R;
+import org.chromium.chrome.browser.BraveRewardsHelper;
+import org.chromium.chrome.browser.BraveRewardsNativeWorker;
+import org.chromium.chrome.browser.BraveRewardsObserver;
+import org.chromium.chrome.browser.BraveRewardsSiteBannerActivity;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.Locale;
+
+public class BraveRewardsTipConfirmationFragment extends Fragment implements BraveRewardsObserver {
+    private BraveRewardsNativeWorker mBraveRewardsNativeWorker;
+    private int mStatus;
+    private String mPublisherName;
+    private boolean mMonthlyContribution;
+    private TextView mNextTipDateText;
+    private TextView mNextTipDateValue;
+
+    public static BraveRewardsTipConfirmationFragment newInstance(
+            int status, String name, double amount, boolean isMonthly) {
+        BraveRewardsTipConfirmationFragment fragment = new BraveRewardsTipConfirmationFragment();
+        Bundle args = new Bundle();
+        args.putDouble(BraveRewardsSiteBannerActivity.AMOUNT_EXTRA, amount);
+        args.putBoolean(BraveRewardsSiteBannerActivity.IS_MONTHLY_CONTRIBUTION, isMonthly);
+        args.putInt(BraveRewardsSiteBannerActivity.STATUS_ARGS, status);
+        args.putString(BraveRewardsSiteBannerActivity.NAME_ARGS, name);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(
+            LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        mBraveRewardsNativeWorker = BraveRewardsNativeWorker.getInstance();
+        mBraveRewardsNativeWorker.AddObserver(this);
+        View view = inflater.inflate(R.layout.brave_rewards_tip_confirmation, container, false);
+        if (getArguments() != null) {
+            double amount = getArguments().getDouble(BraveRewardsSiteBannerActivity.AMOUNT_EXTRA);
+            mMonthlyContribution = getArguments().getBoolean(
+                    BraveRewardsSiteBannerActivity.IS_MONTHLY_CONTRIBUTION);
+            int status = getArguments().getInt(BraveRewardsSiteBannerActivity.STATUS_ARGS, -1);
+            String publisherName =
+                    getArguments().getString(BraveRewardsSiteBannerActivity.NAME_ARGS);
+
+            initView(view, status, publisherName, amount, mMonthlyContribution);
+        }
+        return view;
+    }
+
+    @SuppressLint("SetTextI18n")
+    private void initView(
+            View view, int status, String name, double amount, boolean monthlyContribution) {
+        mStatus = status;
+        mPublisherName = name;
+        TextView tipAmountTextView = view.findViewById(R.id.amount_sent);
+        tipAmountTextView.setText(amount + " BAT");
```

Comment:
nit: noticed this twice, prefer string template

Reply:
used one constant